extensions/azurePublishNew/src/node/constants.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export const AzureResourceTypes = {
+  APP_REGISTRATION: 'appRegistration',
```

Comment:
IMHO, I don't know why we keep using UPPER_CASE like we're a C++ team writing header files 😜

Reply:
I agree with Geoff + I think this should be a union type.